grpc-api/src/main/java/io/deephaven/grpc_api/session/ExportTicketResolver.java

```diff
@@ -29,7 +34,25 @@ public String getLogNameFor(ByteBuffer ticket) {
 
     @Override
     public Flight.FlightInfo flightInfoFor(final Flight.FlightDescriptor descriptor) {
-        // sessions do not participate in resolving flight descriptors
+        final SessionState session = sessionService.getOptionalSession();
+        if (session != null) {
+            final SessionState.ExportObject<?> export = session.getExportIfExists(ExportTicketHelper.descriptorToExportId(descriptor));
+            if (export != null && export.tryRetainReference()) {
+                try {
+                    //noinspection SynchronizationOnLocalVariableOrMethodParameter
+                    synchronized (export) {
+                        if (export.getState() == ExportNotification.State.EXPORTED) {
+                            final Object realExport = export.get();
+                            if (realExport instanceof Table) {
+                                return TicketRouter.getFlightInfo((Table) realExport, descriptor, ExportTicketHelper.descriptorToTicket(descriptor));
```

Comment:
If we alllow `getFlightInfo`, shouldn't we also allow `listFlights` (`forAllFlightInfo`)?

Reply:
It isn't here based on our past discussions. You convinced me that sharing information of temporary exports might not be as helpful, even if it is low-hanging and easy to implement. That is, the export mechanism is controlled by the client via creating and releasing exports, but listFlights is likely to only be used for clients that don't have the smarts to participate in our export system.

I am fine either way, because implementation cost is low -- but you had a valid point when we last discussed the idea.

Reply:
I don't actually disagree with past-Ryan. I'm just not loving the lack of symmetry here. What's the use-case for supporting `getFlightInfo` on an export ticket?

Reply:
It's to support flight's getFlightInfo and getSchema from ticket. It would seem strange to me that the only way to acquire the Schema for a ticket is via a DoGet (as the first msg). And then at that point, it was most consistent / convenient for the API to return FlightInfo if asked for it.